gvc.go

```diff
@@ -133,74 +185,52 @@ func cleanup(path string, opts options) error {
 		if lastVendorPath == "" {
 			lastVendorPath = localPath
 		}
+		lastVendorPathDir := filepath.Dir(lastVendorPath)
 
 		keep := false
+
 		for _, name := range pkgList {
-			// If the file's parent directory is a needed package, keep it.
-			if !info.IsDir() && filepath.Dir(lastVendorPath) == name {
-				if opts.onlyCode {
-					if opts.noTests && strings.HasSuffix(path, "_test.go") {
-						keep = false
-						continue
-					}
-					for _, suffix := range codeSuffixes {
-						if strings.HasSuffix(path, suffix) {
-							keep = true
-						}
-					}
-					// Match keep patterns
-					for _, keepPattern := range opts.keepPatterns {
-						ok, err := doublestar.Match(keepPattern, lastVendorPath)
-						// TODO(sgotti) if a bad pattern is encountered stop here. Actually there's no function to verify a pattern before using it, perhaps just a fake match at the start will work.
-						if err != nil {
-							return fmt.Errorf("bad pattern: %q", keepPattern)
-						}
-						if ok {
-							keep = true
-						}
-					}
-				} else {
-					keep = true
-				}
+			// If a directory is a needed package then keep it
+			keep = keep || info.IsDir() && name == lastVendorPath
+
```

Comment:
Yes, I figured the comparisons were all operating on things already in memory, so there wouldn't be a huge performance hit, and I felt it made the code easier to follow. It should be pretty easy to add the breaks back if needed.